p2p/host/relay/autorelay.go

```diff
@@ -245,6 +248,10 @@ func (ar *AutoRelay) connect(ctx context.Context, pi peer.AddrInfo) bool {
 }
 
 func (ar *AutoRelay) discoverRelays(ctx context.Context) ([]peer.AddrInfo, error) {
+	if len(ar.static) > 0 {
```

Comment:
I don't think setting static relays should be exclusive from dynamic relay discovery. You probably want static relays for a speedier bootup, but as you dynamically discover relays, you should preferentially switch to those.

Reply:
I see that @albrow expected the same thing: https://github.com/libp2p/go-libp2p/pull/705#issuecomment-522708344

Reply:
This defeats the purpose; the main impetus for the static relay configuration is to avoid hitting the DHT _altogether_.
We might want to leverage passive discovery with `CanHop` in the relay module, but I don't think it's a good idea to have both static relays _and_ hit the dht.